examples/amm/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+use std::convert::Infallible;
+
+use async_graphql::{scalar, Request, Response};
+use fungible::AccountOwner;
+use linera_sdk::base::{Amount, ArithmeticError, ContractAbi, ServiceAbi};
+use linera_views::views::ViewError;
+use matching_engine::Parameters;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+pub struct AmmAbi;
+
+impl ContractAbi for AmmAbi {
+    type InitializationArgument = ();
+    type Parameters = Parameters;
+    type Operation = Operation;
+    type ApplicationCall = ApplicationCall;
+    type Message = Message;
+    type SessionCall = ();
+    type Response = ();
+    type SessionState = ();
+}
+
+impl ServiceAbi for AmmAbi {
+    type Query = Request;
+    type QueryResponse = Response;
+    type Parameters = Parameters;
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum OperationType {
+    Swap {
+        owner: AccountOwner,
+        input_token_idx: u32,
+        input_amount: Amount,
+    },
+    AddLiquidity {
+        owner: AccountOwner,
+        max_token0_amount: Amount,
+        max_token1_amount: Amount,
+    },
+    RemoveLiquidity {
+        owner: AccountOwner,
+        input_token_idx: u32,
+        input_amount: Amount,
+    },
+}
+
+scalar!(OperationType);
+
+/// Operations that can be sent to the application.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Operation {
+    ExecuteOperation { operation: OperationType },
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum Message {
+    ExecuteOperation { operation: OperationType },
```

Comment:
@andresilva91 Please define `Message` and `ApplicationCall` to only include `Swap`.